containers/EditFish.js

```diff
@@ -1,10 +1,82 @@
-import React, { Component } from "react";
-import { StyleSheet, Text, View, Button, FlatList } from "react-native";
-
-export default function EditFish({ navigation }) {
-  return (
-    <View>
-      <Text>Edit Fish</Text>
-    </View>
-  );
-};
+import React, { Component }  from 'react';
+import FishForm from '../components/FishForm';
+import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+import { setCurrentFish, clearFishEntry } from '../actions'
+import { addUpdatedFish } from "../ApiCalls";
+import { connect } from "react-redux";
+
+class EditFish extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            selectedFish: null
+        }
+    }
+
+    componentDidMount = () => {
+        const selectedFish = this.props.currentFish.find(fish => fish.id == this.props.selectedFishId)
+        this.setState({selectedFish: selectedFish})
+        this.setCurrentFish(selectedFish)
+    }
+
+    setCurrentFish = (selectedFish) => {
+        this.props.setCurrentFish(selectedFish.attributes)
+    }
+
+    updateFish = () => {
+        addUpdatedFish(this.state.selectedFish)
+    }
+
+    render() {
+       return (
+        <View style={styles.container}>
+            <FishForm />
+            <View style={{flex: 1, flexDirection: 'row', justifyContent: 'center'}}>
+                <TouchableOpacity style={styles.button}
+                    onPress = {() => {
+                    this.props.clearFishEntry()
+                    this.props.navigation.navigate('FishCaught');
+                    }}>
+                    <Text style={styles.button}>Cancel</Text>
+                </TouchableOpacity>
+                
+                <TouchableOpacity style={styles.button}
+                    onPress = {() => {
+                    this.props.clearFishEntry()
+                    this.updateFish()
+                    this.props.navigation.navigate('FishCaught');
+                    }}>
+                    <Text style={styles.button}>Submit</Text>
+                </TouchableOpacity>
```

Comment:
We will need to add the method where on the submission of an edited fish, it should automatically update the Fish store too so that the user doesn't have to reload the application to see the changes.  This feature is related to my current PR of updating the Fish Store on submit.  We can implement it in a future update.

Reply:
Thanks for pointing this out! I can work on that on the branch with the delete functionality and I'll let you know if I have any questions.